sst/SimengCoreWrapper.cc

```diff
@@ -127,52 +117,52 @@ bool SimengCoreWrapper::clockTick(SST::Cycle_t current_cycle) {
   }
 }
 
+std::vector<std::string> SimengCoreWrapper::splitArgs(std::string argString) {
```

Comment:
Whilst this is a good idea generally, maybe we should re-evaluate the custom delimiters. If we deem `[` or `]` to be valid arguments then we need to support all possible permutations. It may be worth us outputting what workload and args are being used in the sim. Perhaps inline or below the "Running in x mode"

Reply:
I agree, outputting args is a good idea.  Do you think having an escape character for `[ ]` would solve our issue or will it make it too complicated?

Reply:
Yeh I think so. Then we can have errored states for []] but not [\]]

Reply:
Sorry, I don't see how both those are different.

Reply:
updated comment

Reply:
I think the following should give us the desired functionality:
```
std::vector<std::string> SimengCoreWrapper::splitArgs(std::string argString) {
  int brCount = 0;
  std::string newArg = "";
  std::vector<std::string> arguments = {};

  // Using a custom delimiter, split the argString string into individual
  // arguments and collate in a vector
  for (int c = 0; c < argString.length(); c++) {
    // Recored starting delimiter
    if (argString.at(c) == '[') {
      brCount++;
    } else if (argString.at(c) == ']') {
      // Ensure ending delimiter has a matching start
      if (brCount <= 0) {
        std::cerr << "Invalid argument format, mismatch on delimiters"
                  << std::endl;
        exit(1);
        return {};
      } else {
        arguments.push_back(newArg);
        newArg = "";
      }
      brCount--;
    } else if (argString.at(c) == '\\') {
      // Identified escape character, read next character as an argument
      // character
      c++;
      newArg += argString.at(c);
    } else {
      newArg += argString.at(c);
    }
  }

  // Ensure all starting delimiters have been matched
  if (brCount != 0) {
    std::cerr << "Invalid argument format, unmatched delimiters" << std::endl;
    exit(1);
    return {};
  }

  return arguments;
}

Reply:
Omitting the fix for outputting workload information as it will be contained in an upcoming PR

Reply:
This looks good! 
Is there any way we could test it? (if we need do)

Reply:
The inner if can be refactored like this:
```
 else if (argString.at(c) == ']') {
      // Ensure ending delimiter has a matching start
      if (brCount <= 0) {
        std::cerr << "Invalid argument format, mismatch on delimiters"
                  << std::endl;
        exit(1);
      } 
      arguments.push_back(newArg);
      newArg = "";
      brCount--;
    }
```

Reply:
> This looks good! Is there any way we could test it? (if we need do)

Other than manually, I don't know where we'd test it other than your framework